src/main/java/com/google/sps/servlets/SpoonacularCombinedServlet.java

```diff
@@ -38,6 +38,7 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String query = request.getParameter("dishName");
     Client client = ClientBuilder.newClient();
+    query = query.replaceAll(" ", "+");
```

Comment:
You might want to use  URLEncoder.encode(query, "UTF-8") instead. There are also other characters need to be escaped for url: https://www.w3schools.com/tags/ref_urlencode.ASP